src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs

```diff
@@ -121,6 +124,40 @@ impl CloudHypervisorInner {
         Ok(())
     }
 
+    async fn handle_vfio_device(&mut self, device: &VfioDevice) -> Result<()> {
+        // A device with multi-funtions, or a IOMMU group with one more
+        // devices, the Primary device is selected to be passed to VM.
+        // And the the first one is Primary device.
+        // safe here, devices is not empty.
+        let primary_device = device.devices.first().unwrap().clone();
```

Comment:
Could you use `?` rather than `.unwrap()` as the function returns a `Result`?

Reply:
Fixed